### PR TITLE
Compile error instead of falling back to C for unsupported calling conventions

### PR DESCRIPTION
**Pull request by tadeokondrak**
Closes https://github.com/ziglang/zig/issues/6536.

### CODE REVIEW

**Comment by tadeokondrak** [comment]
Trying to build the self-hosted compiler for `x86_64-windows-gnu` on my machine seems to cause lld to call _exit somewhere during linking. Not sure why.

File: lib/std/os/windows/bits.zig [hunk 1]
```diff
@@ -24,6 +24,11 @@ pub const STD_OUTPUT_HANDLE = maxInt(DWORD) - 11 + 1;
 /// The standard error device. Initially, this is the active console screen buffer, CONOUT$.
 pub const STD_ERROR_HANDLE = maxInt(DWORD) - 12 + 1;
 
+pub const WINAPI: builtin.CallingConvention = if (builtin.arch == .i386)
+    .Stdcall
+else
+    .C;
+
```
Comment (Rocknest) [inline lib/std/os/windows/bits.zig:31]: What? Why? stdcall is always used except on arm

Comment (tadeokondrak) [reply lib/std/os/windows/bits.zig:31]: No, see the current code: https://github.com/ziglang/zig/blob/eb33394d14e29600d36280b6797de0d8f40076c1/src/stage1/codegen.cpp#L165-L168

stdcall doesn't exist on x86_64.

Comment (Rocknest) [reply lib/std/os/windows/bits.zig:31]: Oh right! M$VC ignores even cdecl on x86_64 and uses its own 'x64' callconv.

File: src/stage1/analyze.cpp [hunk 2]
```diff
@@ -1861,6 +1861,58 @@ ZigType *get_auto_err_set_type(CodeGen *g, ZigFn *fn_entry) {
...
+        case CallingConventionNaked:
+        case CallingConventionAsync:
+            break;
+        case CallingConventionInterrupt:
+            if (g->zig_target->arch != ZigLLVM_x86
+                && g->zig_target->arch != ZigLLVM_x86_64
+                && g->zig_target->arch != ZigLLVM_avr
+                && g->zig_target->arch != ZigLLVM_msp430)
+            {
+                allowed_platforms = "x86, x86_64, AVR, and MS430";
```
Comment (LemonBoy) [inline src/stage1/analyze.cpp:1880]:
```suggestion
                allowed_platforms = "x86, x86_64, AVR, and MSP430";
```

**Comment by LemonBoy** [comment]
The old behaviour was modeled after clang as it ignores (and afair gives a warning) whenever it encounters a invalid cc for the given target. But giving an error is also fine and will force people to use the correct cc, so +1 from me.

**Comment by Rocknest** [comment]
How do i use SysV callconv when target os = windows?

**Comment by tadeokondrak** [comment]
> How do i use SysV callconv when target os = windows?

I don't think this is possible but it should be. I don't think it's related to this PR though.

**Comment by tadeokondrak** [comment]
Still don't know why this is failing, so closing until #3825 is fixed

**Comment by LemonBoy** [comment]
> Still don't know why this is failing, so closing until #3825 is fixed

You should be able to see LLD's fatal errors by modifying `ZigLLDLink` to use `llvm::errs()` and `llvm::outs()` instead of `diag_stderr` and `diag_stdout`, the former are two unbuffered streams going to the terminal while the latter are the fake ones Zig setups in order to collect the error messages.
When LLD calls `fatal()` the whole Zig process is terminated and the error messages get stuck in the diag streams.

**Pull request closed without merging**